File: 27.remove-element.go
/*
 * @lc app=leetcode id=27 lang=golang
 *
 * [27] Remove Element
 */

// @lc code=start
func removeElement(nums []int, val int) int {
	cnt, ptr := 0, 0
	for _, v := range nums {
		if v != val {
			nums[ptr] = v
			ptr++
			cnt++
		}
	}
	return cnt
}

// @lc code=end

File: java/27.remove-element.java
/*
 * @lc app=leetcode id=27 lang=java
 *
 * [27] Remove Element
 */
class Solution {
    public int removeElement(int[] nums, int val) {
        if (nums == null || nums.length == 0) return 0;
        int mainIdx = 0;
        for (int runningIdx = 0; runningIdx < nums.length; ++runningIdx) {
            if (nums[runningIdx] != val) {
                if (runningIdx != mainIdx) {
                    nums[mainIdx] = nums[runningIdx];
                }
                mainIdx++;
            }
        }
        return mainIdx;
    }
}

